### DISCUSSION TITLE
Performance of ThorVG

### DISCUSSION THREAD

**Discussion by xland** [thorvg/thorvg#1815] [Q&A] [answered]
I'm not trying to start a war.
I just want to know are they right, from a technical point of view?

> I think Blend2D would beat ThorVG SW renderer significantly, because ThorVG is basically using FreeType code, which was not written for performance. I would assume ThorVG and Qt to give you similar performance (ThorVG maybe worse in some cases as it has very limited SIMD use, Qt at least has SIMD composition).  

[https://github.com/blend2d/blend2d/issues/3#issuecomment-1113168818](https://github.com/blend2d/blend2d/issues/3#issuecomment-1113168818)

> I agree. ThorVG was created to run on low resource devices, like smart TVs, so performance was not the top priority. However I would be interested to see exactly what the difference is :)

[https://github.com/blend2d/blend2d/issues/3#issuecomment-1113197948](https://github.com/blend2d/blend2d/issues/3#issuecomment-1113197948)

Performance of Blend2D:
https://blend2d.com/performance.html

**Comment by hermet** [comment, marked as answer]
@xland Hello, that argument does make sense. Previously, a performance test comparing ThorVG with Cairo was conducted unofficially. And ThorVG has since been introduced, Blend2D has not yet been involved in such comparisons. However, with the practical products, ThorVG has undergone optimizations compared to its original state, including the addition of threading and fine-grained fast-track drawing, etc. and we'll continue to explore further optimization possibilities. While ThorVG's software renderer is primarily designed for low-end devices, we have the flexibility to switch to backends such as GL or WGPU if performance becomes a top priority. To make informed decisions, we currently need accurate benchmarking data that includes Blend2D and other relevant tools later.

**Comment by hermet** [comment]
@xland mark as answer?

**Reply by xland** [reply]
Sorry, I forget this.
